src/lat/lattice-functions.h

```diff
@@ -377,6 +377,37 @@ void ComposeCompactLatticeDeterministic(
     fst::DeterministicOnDemandFst<fst::StdArc>* det_fst,
     CompactLattice* composed_clat);
 
+/// This function computes and returns mapping from the pair 
+/// (frame-index, pdf-id) to the acoustic score for that pdf-id for that 
+/// frame-index in the lattice. 
+/// This function is useful for retaining the acoustic scores in a 
+/// non-compact lattice after a process like determinization where the 
+/// frame-level acoustic scores are typically lost.
+/// The function ReplaceAcousticScoresFromMap is used to restore the 
+/// acoustic scores computed by this function.
+///
+///   @param [in] lat   Input lattice. Expected to be top-sorted. Otherwise the 
+///                     function will crash. 
+///   @param [out] acoustic_scores  Pointer to a map where the mapping from the
```

Comment:
The documentation doesn't seem to be consistent with the function signature: you say it is a map
to acoustic score, but it returns a pair.